python/ambassador/envoy/v2/v2cluster.py

```diff
@@ -51,6 +51,17 @@ def __init__(self, config: 'V2Config', cluster: IRCluster) -> None:
         cmap_entry = cluster.clustermap_entry()
         if Config.legacy_mode or (cmap_entry['kind'] == 'KubernetesServiceResolver'):
             ctype = cluster.type.upper()
+            # For now we are only allowing Logical_dns for the cluster since it is similar enough to strict_dns that we dont need any other config changes
+            # It should be easy to add the other dns_types here in the future if we decide to support them
+            allowedDnsTypes = ['STRICT_DNS', 'LOGICAL_DNS']
+            isTypeAllowed = False
+            for dnsType in allowedDnsTypes:
+                if ctype == dnsType:
+                    isTypeAllowed = True
+                    break
+            if not isTypeAllowed:
```

Comment:
This can just be

```
if dnsType not in [ "STRICT_DNS", "LOGICAL_DNS" ]:
    # scream here
```

The `in` operator, when applied to a list, does a linear search, so it's a poor choice when you have a really long list... but with such a small number, it'll be literally as fast as looking into a Python map, and way faster than doing the iteration in Python. 😉

Reply:
Good feedback, thanks!